File: 30 Days of Code/Day 8: Dictionaries and Maps/solution.cpp
// https://www.hackerrank.com/challenges/30-dictionaries-and-maps/problem

#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
#include <map>
using namespace std;


int main() {
    int n;
    cin >> n;

    map<string, int> phonebook;
    while (n--) {
        string name;
        int phone;
        cin >> name >> phone;

        phonebook[name] = phone;
    }

    string name;
    while (cin >> name) {
        if (phonebook.find(name) == phonebook.end()) {
            cout << "Not found\n";
        } else {
            cout << name << "=" << phonebook[name] << "\n";
        }
    }
    return 0;
}

File: 30 Days of Code/Day 8: Dictionaries and Maps/solution.py
# https://www.hackerrank.com/challenges/30-dictionaries-and-maps/problem

import sys

n = int(input())

phonebook = dict()

for _ in range(n):
    name, phone = input().split()
    phonebook[name] = phone

names = sys.stdin.readlines()
for name in names:
    name = name.strip()
    phone = phonebook.get(name, None)

    if phone == None:
        print("Not found")
    else:
        print(name + "=" + phone)
